### ISSUE TITLE
Defining sample weights or class weights with MultiClassOneVsAll loss causes Segmentation Fault

### ISSUE THREAD

**Issue by vBLFTePebWNi6c** [catboost/catboost#109]
Here is the code to reproduce problem:
```
from catboost import CatBoostClassifier
cat = CatBoostClassifier(loss_function='MultiClassOneVsAll', 
class_weights=class_weights # <-- this
)

cat.fit(X=X_train, y=y_train, 
sample_weight=sample_weight # <-- or this
) 
```
class_weights and sample_weight are non empty float lists

**Comment by vitekmel** [comment]
Thanks! It was a bug, should be fixed in next version.

**Comment by vitekmel** [comment]
Should be fixed now, could you please recheck?

**Comment by vBLFTePebWNi6c** [comment]
@vitekmel works fine. Thanks!

**Comment by WaelKarkoub** [comment]
@vitekmel everytime I run .predict(), it randomly crashes and i get the error 

> segmentation fault (core dumped)  /usr/bin/python3.5

The core dump is below

> 0x00007fffbb923510 in TStaticCtrProvider::~TStaticCtrProvider() ()
>    from /home/user/.local/lib/python3.5/site-packages/catboost/_catboost.so

is there a possible fix for this?

**Comment by sab** [comment]
@WaelKarkoub have you tried latest CatBoost version?

**Comment by WaelKarkoub** [comment]
@sab Yeah it's the latest version, I tried re-installing it as well.

**Comment by WaelKarkoub** [comment]
@sab I also tried predict_proba and still gives me the same error

**Comment by kizill** [comment]
@WaelKarkoub , tell me about your model - you are using multiclass model with categorical features&

**Comment by WaelKarkoub** [comment]
@kizill im using multiclass one vs all with numerical features and the labels used to be categorical but converted them to numerical labels.
'     def CAT_train(self,samples,labels,model_output_path):
        d,labels = encoder(labels)
        # samples.astype(int)
        X_train, X_test, y_train, y_test = cross_validation.train_test_split(samples, labels, test_size = 0.2)
        
        cat = CatBoostClassifier(iterations=100, learning_rate = 0.01,loss_function='MultiClass',eval_metric='AUC', classes_count = 4)
        cat.fit(X_train, y_train)

        joblib.dump(cat, model_output_path)
        joblib.dump(d, 'dictionary.sav')
    
        y_predict=cat.predict(X_test)
    
        print("accuracy = {}".format(accuracy_score(y_test, y_predict)))        

    def CAT_predict(self,sample, catFile):
        cat = joblib.load(catFile)
        d = joblib.load('dictionary.sav')
        prediction = cat.predict_proba(sample)
        print(sample)
        # new_pred = ['k']*len(prediction)
        # for i, k in enumerate(prediction):
        #     for key, value in d.items():
        #         if value == k:
        #             new_pred[i] = key
        return prediction'
Above is the code that im using. Encoder is a helper function. I'm running this code on a video and it crashes randomly and not on a specific frame.

**Comment by kizill** [comment]
@WaelKarkoub i'll try to reproduce problem.

**Comment by WaelKarkoub** [comment]
@kizill were you able to reproduce my problem?
